src/components/inputs/file/file.component.tsx

```diff
@@ -20,6 +21,29 @@ const File: React.FC<FormFieldInputProps> = ({ field, value, setFieldValue }) =>
   const [dataSource, setDataSource] = useState<DataSourceType>(null);
   const { sessionMode } = useFormProviderContext();
 
+  const deleteAttachment = useCallback(() => {
+    if (value && value.id) {
+      deleteAttachmentPermanently(value.id, new AbortController())
```

Comment:
The abort controller isn't being correctly used here (I know we keep making this same mistake across the codebase). You want to emit an abort signal when the component unmounts.

```tsx

// declare controller
const abortController = new AbortController();

useEffect(() => {
  // emit an abort signal when the component unmounts 
   return () => abortController.abort();
}, []);

const handleDeleteAttachment =  useCallback(() => {  
   // logic that consumes the controller

}, [value, setFieldValue, t]);

```

Reply:
@TrevorAntony a better way of approaching the above is by using [useAbortController](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-react-utils/src/useAbortController.ts#L25) defined in core:


```tsx

  const abortController = useAbortController();

  const handleDeleteAttachment =  useCallback(() => {  
     // logic that consumes the controller
  
  }, [value, setFieldValue, t]);

```